src/neuroglancer/util/special_protocol_request.ts

```diff
@@ -67,6 +91,12 @@ export function parseSpecialUrl(url: string, credentialsManager: CredentialsMana
         credentialsProvider: getNgauthCredentialsProvider(credentialsManager, `https://${u.host}`, u.path),
         url: 'gs+xml:/' + u.path,
       };
+      case 'middleauth+https':
```

Comment:
Now that there are a decent number of protocols, it might be nicer to add a registry mechanism so that they don't all have to be defined in this file.  That is also something I could take care of later after merging though.